src/components/WorkOrder/EditInventoryItemsFields.js

```diff
@@ -0,0 +1,155 @@
+          return { label: item.identifier, value: item.id };
+        });
+        this._isMounted && this.setState({ itemOptions });
+      });
+  };
+
+  handleConditionChange = e => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_CONDITION] = e.target.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleQuantityChange = e => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_QUANTITY] = e.target.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleSourceChange = option => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_SOURCE] = option.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleCommentChange = e => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_COMMENT] = e.target.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  render() {
+    const existingFormData = this.state.existingFormData;
+    return existingFormData ? (
+      <>
+        <div className="form-group">
+          <label htmlFor={FIELDS.WORK_ORDER_INVENTORY_ITEMS}>
+            Inventory Item
+          </label>
+          <Select
+            className="basic-single"
+            classNamePrefix="select"
+            defaultValue={convertKnackResponseObjectToSelectFormOption(
+              existingFormData[FIELDS.WORK_ORDER_EDIT_INVENTORY_ITEM][0]
+            )}
+            isClearable
+            isSearchable
+            name={FIELDS.WORK_ORDER_INVENTORY_ITEMS}
+            options={this.state.itemOptions}
+            onChange={this.props.handleInventoryItemChange.bind(
+              this,
+              FIELDS.WORK_ORDER_INVENTORY_ITEMS
+            )}
+            required // Prevent blank item from adding to DB since Knack does not require these fields
+          />
+        </div>
+        <div className="form-group">
+          <label htmlFor={FIELDS.WORK_ORDER_ITEM_QUANTITY}>Quantity</label>
+          <input
+            className="form-control"
+            id={FIELDS.WORK_ORDER_ITEM_QUANTITY}
+            name={FIELDS.WORK_ORDER_ITEM_QUANTITY}
+            type="number"
+            defaultValue={existingFormData[FIELDS.WORK_ORDER_ITEM_QUANTITY]}
+            onChange={this.handleQuantityChange}
+            required // Prevent blank item from adding to DB since Knack does not require these fields
+          />
+        </div>
+        <div className="form-group">
+          <label htmlFor={FIELDS.WORK_ORDER_ITEM_SOURCE}>Source</label>
+          <Select
+            className="basic-single"
+            classNamePrefix="select"
+            id={FIELDS.WORK_ORDER_ITEM_SOURCE}
+            name={FIELDS.WORK_ORDER_ITEM_SOURCE}
+            defaultValue={convertKnackResponseStringToSelectFormOption(
+              existingFormData[FIELDS.WORK_ORDER_ITEM_SOURCE]
+            )}
+            options={getWorkTypeScheduledWorkOptions([
+              "Warehouse Inventory",
+              "Working Stock",
+              "Truck Stock",
+              "Other",
```

Comment:
guessing there wasn't a way to get these options programmatically from Knack via their API?

Reply:
There wasn't a way that I could see to get these values from the Knack API. There isn't an API call exposed for these values when inspecting this form in the Knack app (like there is for the signals). It makes me really curious about where this data is coming from in Knack. Maybe we could take a look at this together at some point.